src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -1908,9 +1908,15 @@ protected override void BeginProcessing()
 
                 if (_UseNewEnvironment)
                 {
-                    startInfo.EnvironmentVariables.Clear();
-                    LoadEnvironmentVariable(startInfo, Environment.GetEnvironmentVariables(EnvironmentVariableTarget.Machine));
-                    LoadEnvironmentVariable(startInfo, Environment.GetEnvironmentVariables(EnvironmentVariableTarget.User));
+                    // Use environment variables from saved at startup dictionary
+                    var envVariables = startInfo.EnvironmentVariables;
+                    envVariables.Clear();
+                    IDictionaryEnumerator e = System.Management.Automation.Runspaces.EarlyStartup.InitialEnvironmentVariables.GetEnumerator();
```

Comment:
Just curious, why did you use an enumerator object explicitly rather than:
```
foreach (DictionaryEntry entry in EarlyStartup.InitialEnvironmentVariables)
{
    ...
}
```
?

Reply:
I copied the code from CoreFX.

Fixed.